src/config.rs

```diff
@@ -158,40 +158,45 @@ impl TitanCfConfig {
     }
 }
 
-fn get_background_job_limit(
-    default_background_jobs: i32,
-    default_background_flushes: i32,
-    default_sub_compactions: u32,
-    default_background_gc: i32,
-) -> (i32, i32, u32, i32) {
+#[derive(Clone, Copy, Debug)]
+struct BackgroundJobLimits {
+    max_background_jobs: u32,
+    max_background_flushes: u32,
+    max_sub_compactions: u32,
+    max_titan_background_gc: u32,
+}
+
+fn get_background_job_limits(defaults: BackgroundJobLimits) -> BackgroundJobLimits {
     let cpu_num = SysQuota::new().cpu_cores_quota();
     // At the minimum, we should have two background jobs: one for flush and one for compaction.
     // Otherwise, the number of background jobs should not exceed cpu_num - 1.
     // By default, rocksdb assign (max_background_jobs / 4) threads dedicated for flush, and
     // the rest shared by flush and compaction.
-    let max_background_jobs: i32 =
-        cmp::max(2, cmp::min(default_background_jobs, (cpu_num - 1.0) as i32));
-    // Scale flush threads proportionally to background jobs. Also make sure the number of flush
+    let max_background_jobs = cmp::max(
+        2,
+        cmp::min(defaults.max_background_jobs, (cpu_num - 1.0) as u32),
+    );
+    // Scale flush threads proportionally to cpu cores. Also make sure the number of flush
     // threads doesn't exceed total jobs.
-    let max_background_flushes: i32 = cmp::min(
-        cmp::max(default_background_flushes, max_background_jobs / 4),
+    let max_background_flushes = cmp::min(
+        cmp::min(defaults.max_background_flushes, (cpu_num as u32) / 4),
```

Comment:
The old constraint shouldn't be dropped, which is reserving at least 2 flush workers even when `cpu_num` is smaller than 8.

Reply:
updated. Adding a min_background_flushes param to enforce that.